src/dune_rules/odoc.ml

```diff
@@ -479,10 +583,13 @@ let setup_lib_html_rules_def =
   end in
   let f (sctx, lib, requires) =
     let ctx = Super_context.context sctx in
-    let odocs = odocs sctx (Lib lib) in
     let pkg = Lib_info.package (Lib.Local.info lib) in
-    List.iter odocs ~f:(setup_html sctx ~pkg ~requires);
-    let html_files = List.map ~f:(fun o -> Path.build o.html_file) odocs in
+    let odoc_artifacts = odoc_artifacts sctx (Lib lib) in
+    List.iter odoc_artifacts ~f:(link_odoc_file sctx ~pkg ~requires);
```

Comment:
I don't think you need this

Reply:
We do need it, `link_odoc_file: unit` because it’s at this point that we link each odoc_file/odoc_artifact, then we do apply `setup_html` on the result.